src/main/kotlin/app/CacheService.kt

```diff
@@ -0,0 +1,85 @@
+@file:Suppress("SqlResolve")
+
+package app
+
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import com.fasterxml.jackson.module.kotlin.readValue
+import org.slf4j.LoggerFactory
+import java.sql.DriverManager
+import java.time.Instant
+import java.time.temporal.ChronoUnit
+
+object CacheService {
+    private const val urlToDb = "jdbc:h2:mem:userinfo"
+    private val log = LoggerFactory.getLogger(CacheService.javaClass)
+    private val objectMapper = jacksonObjectMapper()
+
+    private fun createTableIfAbsent() {
+        val connection = DriverManager.getConnection(urlToDb)
+        val statement = connection.createStatement()
+
+        statement.execute(
+            "CREATE TABLE IF NOT EXISTS userinfo (" +
+            "id VARCHAR2 PRIMARY KEY, " +
+            "timestamp TIMESTAMP, " +
+            "data JSON" +
+            ")"
+        )
+    }
+
+    fun selectJsonFromDb(username: String): String? {
+        val connection = DriverManager.getConnection(urlToDb)
```

Comment:
consider using [HikariCP](https://github.com/brettwooldridge/HikariCP) to keep connections alive instead of creating a new connection per query to minimize latency

Reply:
[HikariCP 5.0.1](https://mvnrepository.com/artifact/com.zaxxer/HikariCP/5.0.1) supports for jdk 11+, so we'll have to pick an older one, right?

Reply:
yeah let's stick with the jdk8 version for now: https://github.com/tipsy/profile-summary-for-github/blob/master/pom.xml#L13

Reply:
Please take a look: [`290d234`](https://github.com/tipsy/profile-summary-for-github/pull/113/commits/290d2344a400aa7fcaff2efc05ed3c267d75e7d9)

Reply:
nice job!